src/main/java/uk/nhs/tis/trainee/notifications/service/HistoryService.java

```diff
@@ -197,14 +197,16 @@ public List<History> findAllFailedForTrainee(String traineeId) {
    * Find all scheduled in-app notifications for the given Trainee.
    *
    * @param traineeId The ID of the trainee to get notifications for.
+   * @param refId The reference ID of the TisReferenceType.
    * @return The found notifications, empty if none found.
    */
-  public List<History> findAllScheduledInAppForTrainee(String traineeId) {
+  public List<History> findAllScheduledInAppForTrainee(String traineeId, String refId) {
```

Comment:
I'd be happier if we included the TisReferenceType as well, to avoid hard-to-find bugs if two records in different tables happen to share the same id value.